File: 2022-05-12-hackathon-4/steps_go_rb/polyglot_step_2.go
//;module Polyglot
//;  def main(&block)
//;    block.call if block
//;  end
//;  def package(_); end
//;  def import(_); end
//;  def string; end
//;  def s(_);end
//;  def var(_); end
//;  def `(str); end
//;  def fmt
//;    Class.new do
//;      def Printf(*args); end
//;    end.new
//;  end
//;  def func(_); end
//;end
//;include Polyglot
package main

import "fmt"

func main() {
	var s string
	s = `package main

import "fmt"

func main() {
	var s string
	s = %c%s%c
	fmt.Printf(s, 96, s, 96)
}
`
	fmt.Printf(s, 96, s, 96)
}

File: 2022-05-12-hackathon-4/steps_kt_rb/polyglot_step_2.kt
//; def val(rb); $rb = rb; end
val rb = '$'.toString() + 'r' + 'b'
val rbCode = """def main(&block)
  block.call
end
def fun(_); end; def println(_); end
def val(str)
  str.define_singleton_method(:format) do |*args|
  end
end
"""

fun main() {
    val template = """//; def val(rb); $rb = rb; end
val rb = '$'.toString() + 'r' + 'b'
val rbCode = ""%c%s%c""

fun main() {
    val template = ""%c%s%c""
    println(template.format('"', rbCode, '"', '"', template, '"'))
}"""
    println(template.format('"', rbCode, '"', '"', template, '"'))
}
